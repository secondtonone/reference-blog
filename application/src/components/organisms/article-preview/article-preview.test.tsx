import renderer from 'react-test-renderer';
import ArticlePreview from '.';
import fixtureAuthors from '~/__fixtures__/authors';

const contentWithText = {
  id: '1',
  title: 'The Ultimate Guide to Google Search Operators and Google Search Commands',
  preview: 'In this guide, weâ€™ll delve deep into the basics of search operators and search commands, helping you understand how to wield these powerful tools effectively and move on to advanced commands and operators.',
  category: {
    name: 'Advanced SEO',
    slug: 'seo/advanced-seo',
    lang: 'en'
  },
  hot: true,
  timeToRead: 5,
  previewImage: '',
  author: fixtureAuthors[0],
  url: 'article',
  publishedAt: 1597908972,
  share: 0,
  background: 'orangeIllust',
  breakpoint: { isPhone: false, isTablet: false, isDesktop: true },
};

const contentWithImage = {
  id: '1',
  title: 'The Ultimate Guide to Google Search Operators and Google Search Commands',
  preview: 'In this guide, weâ€™ll delve deep into the basics of search operators and search commands, helping you understand how to wield these powerful tools effectively and move on to advanced commands and operators.',
  category: {
    name: 'Advanced SEO',
    slug: 'seo/advanced-seo',
    lang: 'en'
  },
  hot: true,
  timeToRead: 5,
  previewImage: '/test-image.jpg',
  author: fixtureAuthors[0],
  url: 'article',
  publishedAt: 1597908972,
  share: 0,
  background: 'orangeIllust',
  breakpoint: { isPhone: false, isTablet: false, isDesktop: true },
};

describe('Match snapshots ArticlePreview', () => {
  it('with text', () => {
    const tree = renderer.create(
      <ArticlePreview
        {...contentWithText}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('with image', () => {
    const tree = renderer.create(
      <ArticlePreview
        {...contentWithImage}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('with text dark', () => {
    const tree = renderer.create(
      <ArticlePreview
        {...contentWithText}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with image dark', () => {
    const tree = renderer.create(
      <ArticlePreview
        {...contentWithImage}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with preview', () => {
    const tree = renderer.create(
      <ArticlePreview
        withPreview
        {...contentWithImage}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('in card', () => {
    const tree = renderer.create(
      <ArticlePreview
        withPreview
        card
        {...contentWithImage}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('horizontal layout', () => {
    const tree = renderer.create(
      <ArticlePreview
        isHorizontal
        {...contentWithImage}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
