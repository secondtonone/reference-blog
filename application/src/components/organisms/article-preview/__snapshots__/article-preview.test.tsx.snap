// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots ArticlePreview horizontal layout 1`] = `
<article
  className="article-preview__ArticleStyled-sc-1u27i03-0 article-preview__ArticlePreviewStyled-sc-1u27i03-1 eJiLdO ckfAuc"
  data-test="article-preview"
>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 djPBEv"
    data-test="container"
  >
    <div
      className="box-adaptive__BoxStyled-sc-g2wy39-0 bXHKQl article-preview__ArticleStyled-sc-1u27i03-0 guofav"
    >
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 wbfkM"
      >
        <a
          className="article-preview__LinkStyled-sc-1u27i03-3 dWYVZU"
          data-test="heading-image-link"
          href="/article"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <img
            alt="The Ultimate Guide to Google Search Operators and Google Search Commands"
            data-test="image"
            height={250}
            src="/test-image.jpg"
            width={310}
          />
        </a>
      </div>
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 cGJWMu"
      >
        <h2
          className="text-content__TextStyled-sc-1stkrsz-0 gHGhqZ"
          data-test="heading-text"
        >
          <a
            data-test="heading-link"
            href="/article"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            The Ultimate Guide to Google Search Operators and Google Search Commands
          </a>
        </h2>
        <div
          className="box-adaptive__BoxStyled-sc-g2wy39-0 dfNHyp"
        >
          <span
            data-test="date"
          >
             
            Aug 20, 2020
          </span>
          <strong
            className="article-preview__CategoryStyled-sc-1u27i03-2 kWXywq"
            data-test="category"
          >
            <a
              data-test="category-link"
              href="/blog/category/seo/advanced-seo"
              onClick={[Function]}
              onMouseEnter={[Function]}
              target="_blank"
            >
              Advanced SEO
            </a>
          </strong>
           
          <span
            data-test="time-to-read"
          >
            5
             
            min read
          </span>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`Match snapshots ArticlePreview in card 1`] = `
<article
  className="article-preview__ArticleStyled-sc-1u27i03-0 article-preview__ArticlePreviewStyled-sc-1u27i03-1 guofav bGodKJ"
  data-test="article-preview"
>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 djPBEv"
    data-test="container"
  >
    <div
      className="box-adaptive__BoxStyled-sc-g2wy39-0 dpoQZX article-preview__ArticleStyled-sc-1u27i03-0 eJiLdO"
    >
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 gmtRFX"
      >
        <a
          className="article-preview__LinkStyled-sc-1u27i03-3 dWYVZU"
          data-test="heading-image-link"
          href="/article"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <img
            alt="The Ultimate Guide to Google Search Operators and Google Search Commands"
            data-test="image"
            height={250}
            src="/test-image.jpg"
            width={310}
          />
        </a>
      </div>
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 dWgrgk"
      >
        <h2
          className="text-content__TextStyled-sc-1stkrsz-0 hqhQTw"
          data-test="heading-text"
        >
          <a
            data-test="heading-link"
            href="/article"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            The Ultimate Guide to Google Search Operators and Google Search Commands
          </a>
        </h2>
        <p
          className="text-content__TextStyled-sc-1stkrsz-0 jGYpim"
          data-test="preview-text"
        >
          In this guide, weâ€™ll delve deep into the basics of search operators and search commands, helping you understand how to wield these powerful tools effectively and move on to advanced commands and operators.
        </p>
        <div
          className="box-adaptive__BoxStyled-sc-g2wy39-0 kqiOeL"
        >
          <span
            data-test="date"
          >
             
            Aug 20, 2020
          </span>
          <strong
            className="article-preview__CategoryStyled-sc-1u27i03-2 kWXywq"
            data-test="category"
          >
            <a
              data-test="category-link"
              href="/blog/category/seo/advanced-seo"
              onClick={[Function]}
              onMouseEnter={[Function]}
              target="_blank"
            >
              Advanced SEO
            </a>
          </strong>
           
          <span
            data-test="time-to-read"
          >
            5
             
            min read
          </span>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`Match snapshots ArticlePreview with image 1`] = `
<article
  className="article-preview__ArticleStyled-sc-1u27i03-0 article-preview__ArticlePreviewStyled-sc-1u27i03-1 eJiLdO ckfAuc"
  data-test="article-preview"
>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 djPBEv"
    data-test="container"
  >
    <div
      className="box-adaptive__BoxStyled-sc-g2wy39-0 bQVcAj article-preview__ArticleStyled-sc-1u27i03-0 guofav"
    >
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 gmtRFX"
      >
        <a
          className="article-preview__LinkStyled-sc-1u27i03-3 dWYVZU"
          data-test="heading-image-link"
          href="/article"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <img
            alt="The Ultimate Guide to Google Search Operators and Google Search Commands"
            data-test="image"
            height={250}
            src="/test-image.jpg"
            width={310}
          />
        </a>
      </div>
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 dWgrgk"
      >
        <h2
          className="text-content__TextStyled-sc-1stkrsz-0 hxfXXq"
          data-test="heading-text"
        >
          <a
            data-test="heading-link"
            href="/article"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            The Ultimate Guide to Google Search Operators and Google Search Commands
          </a>
        </h2>
        <div
          className="box-adaptive__BoxStyled-sc-g2wy39-0 FmeNc"
        >
          <span
            data-test="date"
          >
             
            Aug 20, 2020
          </span>
          <strong
            className="article-preview__CategoryStyled-sc-1u27i03-2 kWXywq"
            data-test="category"
          >
            <a
              data-test="category-link"
              href="/blog/category/seo/advanced-seo"
              onClick={[Function]}
              onMouseEnter={[Function]}
              target="_blank"
            >
              Advanced SEO
            </a>
          </strong>
           
          <span
            data-test="time-to-read"
          >
            5
             
            min read
          </span>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`Match snapshots ArticlePreview with image dark 1`] = `
<article
  className="article-preview__ArticleStyled-sc-1u27i03-0 article-preview__ArticlePreviewStyled-sc-1u27i03-1 eJiLdO ckfAuc"
  data-test="article-preview"
>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 djPBEv"
    data-test="container"
  >
    <div
      className="box-adaptive__BoxStyled-sc-g2wy39-0 bQVcAj article-preview__ArticleStyled-sc-1u27i03-0 guofav"
    >
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 gmtRFX"
      >
        <a
          className="article-preview__LinkStyled-sc-1u27i03-3 dWYVZU"
          data-test="heading-image-link"
          href="/article"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <img
            alt="The Ultimate Guide to Google Search Operators and Google Search Commands"
            data-test="image"
            height={250}
            src="/test-image.jpg"
            width={310}
          />
        </a>
      </div>
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 dWgrgk"
      >
        <h2
          className="text-content__TextStyled-sc-1stkrsz-0 hxfXXq"
          data-test="heading-text"
        >
          <a
            data-test="heading-link"
            href="/article"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            The Ultimate Guide to Google Search Operators and Google Search Commands
          </a>
        </h2>
        <div
          className="box-adaptive__BoxStyled-sc-g2wy39-0 FmeNc"
        >
          <span
            data-test="date"
          >
             
            Aug 20, 2020
          </span>
          <strong
            className="article-preview__CategoryStyled-sc-1u27i03-2 kWXywq"
            data-test="category"
          >
            <a
              data-test="category-link"
              href="/blog/category/seo/advanced-seo"
              onClick={[Function]}
              onMouseEnter={[Function]}
              target="_blank"
            >
              Advanced SEO
            </a>
          </strong>
           
          <span
            data-test="time-to-read"
          >
            5
             
            min read
          </span>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`Match snapshots ArticlePreview with preview 1`] = `
<article
  className="article-preview__ArticleStyled-sc-1u27i03-0 article-preview__ArticlePreviewStyled-sc-1u27i03-1 eJiLdO ckfAuc"
  data-test="article-preview"
>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 djPBEv"
    data-test="container"
  >
    <div
      className="box-adaptive__BoxStyled-sc-g2wy39-0 bQVcAj article-preview__ArticleStyled-sc-1u27i03-0 guofav"
    >
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 gmtRFX"
      >
        <a
          className="article-preview__LinkStyled-sc-1u27i03-3 dWYVZU"
          data-test="heading-image-link"
          href="/article"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <img
            alt="The Ultimate Guide to Google Search Operators and Google Search Commands"
            data-test="image"
            height={250}
            src="/test-image.jpg"
            width={310}
          />
        </a>
      </div>
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 dWgrgk"
      >
        <h2
          className="text-content__TextStyled-sc-1stkrsz-0 efFAUv"
          data-test="heading-text"
        >
          <a
            data-test="heading-link"
            href="/article"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            The Ultimate Guide to Google Search Operators and Google Search Commands
          </a>
        </h2>
        <p
          className="text-content__TextStyled-sc-1stkrsz-0 jGYpim"
          data-test="preview-text"
        >
          In this guide, weâ€™ll delve deep into the basics of search operators and search commands, helping you understand how to wield these powerful tools effectively and move on to advanced commands and operators.
        </p>
        <div
          className="box-adaptive__BoxStyled-sc-g2wy39-0 FmeNc"
        >
          <span
            data-test="date"
          >
             
            Aug 20, 2020
          </span>
          <strong
            className="article-preview__CategoryStyled-sc-1u27i03-2 kWXywq"
            data-test="category"
          >
            <a
              data-test="category-link"
              href="/blog/category/seo/advanced-seo"
              onClick={[Function]}
              onMouseEnter={[Function]}
              target="_blank"
            >
              Advanced SEO
            </a>
          </strong>
           
          <span
            data-test="time-to-read"
          >
            5
             
            min read
          </span>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`Match snapshots ArticlePreview with text 1`] = `
<article
  className="article-preview__ArticleStyled-sc-1u27i03-0 article-preview__ArticlePreviewStyled-sc-1u27i03-1 eJiLdO ckfAuc"
  data-test="article-preview"
>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 djPBEv"
    data-test="container"
  >
    <div
      className="box-adaptive__BoxStyled-sc-g2wy39-0 bQVcAj article-preview__ArticleStyled-sc-1u27i03-0 guofav"
    >
      
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 dWgrgk"
      >
        <h2
          className="text-content__TextStyled-sc-1stkrsz-0 hxfXXq"
          data-test="heading-text"
        >
          <a
            data-test="heading-link"
            href="/article"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            The Ultimate Guide to Google Search Operators and Google Search Commands
          </a>
        </h2>
        <div
          className="box-adaptive__BoxStyled-sc-g2wy39-0 FmeNc"
        >
          <span
            data-test="date"
          >
             
            Aug 20, 2020
          </span>
          <strong
            className="article-preview__CategoryStyled-sc-1u27i03-2 kWXywq"
            data-test="category"
          >
            <a
              data-test="category-link"
              href="/blog/category/seo/advanced-seo"
              onClick={[Function]}
              onMouseEnter={[Function]}
              target="_blank"
            >
              Advanced SEO
            </a>
          </strong>
           
          <span
            data-test="time-to-read"
          >
            5
             
            min read
          </span>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`Match snapshots ArticlePreview with text dark 1`] = `
<article
  className="article-preview__ArticleStyled-sc-1u27i03-0 article-preview__ArticlePreviewStyled-sc-1u27i03-1 eJiLdO ckfAuc"
  data-test="article-preview"
>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 djPBEv"
    data-test="container"
  >
    <div
      className="box-adaptive__BoxStyled-sc-g2wy39-0 bQVcAj article-preview__ArticleStyled-sc-1u27i03-0 guofav"
    >
      
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 dWgrgk"
      >
        <h2
          className="text-content__TextStyled-sc-1stkrsz-0 hxfXXq"
          data-test="heading-text"
        >
          <a
            data-test="heading-link"
            href="/article"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            The Ultimate Guide to Google Search Operators and Google Search Commands
          </a>
        </h2>
        <div
          className="box-adaptive__BoxStyled-sc-g2wy39-0 FmeNc"
        >
          <span
            data-test="date"
          >
             
            Aug 20, 2020
          </span>
          <strong
            className="article-preview__CategoryStyled-sc-1u27i03-2 kWXywq"
            data-test="category"
          >
            <a
              data-test="category-link"
              href="/blog/category/seo/advanced-seo"
              onClick={[Function]}
              onMouseEnter={[Function]}
              target="_blank"
            >
              Advanced SEO
            </a>
          </strong>
           
          <span
            data-test="time-to-read"
          >
            5
             
            min read
          </span>
        </div>
      </div>
    </div>
  </div>
</article>
`;
